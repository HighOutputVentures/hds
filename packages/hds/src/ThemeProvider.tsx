import { ChakraProvider, ChakraProviderProps, extendTheme } from '@chakra-ui/react';
import * as React from 'react';
import Fonts from './Fonts';
import hdsTheme from './theme';

export interface ThemeProviderProps extends ChakraProviderProps {}

const ThemeProvider: React.FC<ThemeProviderProps> = ({ children, theme = {}, ...props }) => {
  return (
    <ChakraProvider theme={extendTheme(hdsTheme, theme)} {...props}>
      <Fonts />

      {children}
    </ChakraProvider>
  );
};

export default ThemeProvider;
