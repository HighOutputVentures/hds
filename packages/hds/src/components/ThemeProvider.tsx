import { ChakraProvider, ChakraProviderProps } from '@chakra-ui/react';
import React, { FC } from 'react';

import { extendTheme } from '../utils/theme.utils';
import Fonts from './Fonts';

export interface ThemeProviderProps extends ChakraProviderProps {}

const ThemeProvider: FC<ThemeProviderProps> = ({ children, ...props }) => {
  return (
    <ChakraProvider {...props} theme={extendTheme(props.theme)}>
      <Fonts />
      {children}
    </ChakraProvider>
  );
};

export default ThemeProvider;
