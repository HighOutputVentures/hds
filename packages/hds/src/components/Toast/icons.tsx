import * as React from 'react';

export function CheckCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm13.36-1.814a.75.75 0 10-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 00-1.06 1.06l2.25 2.25a.75.75 0 001.14-.094l3.75-5.25z"
      />
    </svg>
  );
}

export function ExclamationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
      />
    </svg>
  );
}

export function XMarkIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M5.47 5.47a.75.75 0 011.06 0L12 10.94l5.47-5.47a.75.75 0 111.06 1.06L13.06 12l5.47 5.47a.75.75 0 11-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 01-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 010-1.06z"
      />
    </svg>
  );
}

export function InformationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 01.67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 11-.671-1.34l.041-.022zM12 9a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function ExclamationTriangleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function SpinnerIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="currentColor"
        d="M12,1A11,11,0,1,0,23,12,11,11,0,0,0,12,1Zm0,19a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
        opacity=".25"
      ></path>
      <path
        fill="currentColor"
        d="M12,4a8,8,0,0,1,7.89,6.7A1.53,1.53,0,0,0,21.38,12h0a1.5,1.5,0,0,0,1.48-1.75,11,11,0,0,0-21.72,0A1.5,1.5,0,0,0,2.62,12h0a1.53,1.53,0,0,0,1.49-1.3A8,8,0,0,1,12,4Z"
      >
        <animateTransform
          attributeName="transform"
          dur="0.75s"
          repeatCount="indefinite"
          type="rotate"
          values="0 12 12;360 12 12"
        ></animateTransform>
      </path>
    </svg>
  );
}

export const Number123Icon = React.forwardRef<
  SVGSVGElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="none"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        d="m3 10l2-2v8m4-8h3a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h3m4-8h2.5A1.5 1.5 0 0 1 21 9.5v1a1.5 1.5 0 0 1-1.5 1.5H18h1.5a1.5 1.5 0 0 1 1.5 1.5v1a1.5 1.5 0 0 1-1.5 1.5H17"
      ></path>
    </svg>
  );
});
