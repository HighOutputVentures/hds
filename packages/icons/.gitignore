*.log
.DS_Store
node_modules
.cache
!.vscode
storybook-static
.vercel
