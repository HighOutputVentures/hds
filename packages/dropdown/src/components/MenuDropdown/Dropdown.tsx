import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons';
import {
  Box,
  Button,
  Divider,
  Menu,
  MenuButton,
  MenuGroup,
  MenuItem,
  MenuList,
  PlacementWithLogical,
  UseDisclosureReturn,
  useDisclosure,
} from '@chakra-ui/react';
import { ButtonVariantsTypes } from '@highoutput/hds/src/theme/components/button/variants';
import React, { FC } from 'react';

export interface Items {
  label: string;
  command?: string;
  icon?: JSX.Element;
}
export interface IMenuItemsSingle {
  items: Array<Items>;
}

export interface IMenuItemsGroup {
  items: Array<Array<Items>>;
}
export interface DropdownFieldMainProps {
  menuButtonText?: string;
  profileUrl?: string;
  indicator?: boolean;
  renderHeader?: React.ReactNode;
  __menuTestId?: string;
  __menuButtonTestId?: string;
  __menuListTestId?: string;
  gap?: string;
  placement?: PlacementWithLogical | undefined;
  variant?: ButtonVariantsTypes;
  closeOnSelect?: boolean;
  showDropdownIndicator?: boolean;
  children(context: UseDisclosureReturn): JSX.Element | null;
}
export interface DropdownFieldPropsGroup extends DropdownFieldMainProps {
  group: true;
  items: IMenuItemsGroup['items'];
  hasDivider?: never;
}
export interface DropdownFieldPropsSingle extends DropdownFieldMainProps {
  group: false;
  items: IMenuItemsSingle['items'];
  hasDivider?: boolean;
}

export type DropdownFieldProps =
  | DropdownFieldPropsGroup
  | DropdownFieldPropsSingle;
const Dropdown: FC<DropdownFieldProps> = (props) => {
  const {
    renderHeader,
    closeOnSelect = true,
    showDropdownIndicator = false,
    items,
    gap,
    group,
    hasDivider,
    children,
    __menuTestId,
    __menuButtonTestId,
    __menuListTestId,
    placement,
    variant = 'unstyled',
  } = props;
  const disclosure = useDisclosure();
  return (
    <Menu
      data-testid={__menuTestId ?? 'hds.menu.dropdown'}
      placement={placement}
      closeOnSelect={closeOnSelect}
    >
      <Box w="full">
        <MenuButton
          variant={variant}
          as={Button}
          isActive={disclosure.isOpen}
          rightIcon={
            !disclosure.isOpen && showDropdownIndicator ? (
              <ChevronDownIcon width="20px" height="20px" />
            ) : disclosure.isOpen && showDropdownIndicator ? (
              <ChevronUpIcon width="20px" height="20px" />
            ) : undefined
          }
          data-testid={__menuButtonTestId}
          display="flex"
        >
          {children(disclosure)}
        </MenuButton>

        <MenuList
          boxShadow={
            '0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)'
          }
          marginTop={gap}
          data-testid={__menuListTestId ?? 'hds.menu.list'}
          py={'0px'}
          fontSize={'14px'}
          color="neutrals.900"
        >
          {renderHeader}
          <Divider orientation="horizontal" />
          <MenuGroup>
            {!group &&
              (items as IMenuItemsSingle['items']).map(
                ({ command, label, icon }, idx) => {
                  return (
                    <>
                      <MenuItem command={command} icon={icon}>
                        {label}
                      </MenuItem>
                      {hasDivider && idx !== items.length - 1 && <Divider />}
                    </>
                  );
                }
              )}
            {group &&
              (items as IMenuItemsGroup['items']).map((item) => {
                return item.map(({ label, command, icon }, idx) => {
                  return (
                    <>
                      <MenuItem command={command} icon={icon}>
                        {label}
                      </MenuItem>
                      {idx === item.length - 1 && <Divider />}
                    </>
                  );
                });
              })}
          </MenuGroup>
        </MenuList>
      </Box>
    </Menu>
  );
};

export default Dropdown;
