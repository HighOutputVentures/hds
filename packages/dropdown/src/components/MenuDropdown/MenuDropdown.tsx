import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons';
import {
  Avatar,
  AvatarBadge,
  Box,
  Button,
  Divider,
  HStack,
  Icon,
  Menu,
  MenuButton,
  MenuItemProps,
  MenuList,
  PlacementWithLogical,
  Text,
} from '@chakra-ui/react';
import { ThreeDots } from '@highoutput/hds-icons';
import { ButtonVariantsTypes } from '@highoutput/hds';
import React, { FC } from 'react';

export interface MenuDropdownFieldProps {
  menuType: 'kebab' | 'button' | 'profile' | 'meatball';
  menuButtonText?: string;
  profileUrl?: string;
  indicator?: boolean;
  menuHeader?: {
    profileUrl: string;
    userName: string;
    emailAddress: string;
  };
  menuItems: Omit<MenuItemProps, 'css' | 'style'>;
  gap?: string;
  placement?: PlacementWithLogical | undefined;
  variant?: ButtonVariantsTypes;
}

const MenuDropdown: FC<MenuDropdownFieldProps> = (props) => {
  const {
    menuHeader,
    menuButtonText = '',
    indicator = true,
    menuItems,
    gap,
    profileUrl,
    menuType,
    placement,
    variant = 'unstyled',
  } = props;

  return (
    <Menu data-testid="menu-dropdown" placement={placement}>
      {({ isOpen }) => (
        <Box w="full">
          <MenuButton
            variant={variant}
            as={Button}
            isActive={isOpen}
            rightIcon={
              menuType === 'button' && !isOpen ? (
                <ChevronDownIcon width="20px" height="20px" />
              ) : menuType === 'button' && isOpen ? (
                <ChevronUpIcon width="20px" height="20px" />
              ) : undefined
            }
            data-testid={
              menuType === 'button'
                ? 'menu-button'
                : menuType === 'kebab'
                ? 'menu-kebab'
                : 'menu-profile'
            }
            display="flex"
          >
            {menuType === 'kebab' ? (
              <Icon
                as={ThreeDots}
                width="20px"
                height="20px"
                _active={{ background: 'transparent' }}
                color={isOpen ? ' #344054' : '#98A2B3'}
              />
            ) : menuType === 'profile' ? (
              <Avatar
                src={profileUrl}
                border={isOpen ? '4px solid #F4EBFF' : undefined}
              />
            ) : menuType === 'meatball' ? (
              <Box transform={'rotate(90deg)'}>
                <Icon
                  as={ThreeDots}
                  width="20px"
                  height="20px"
                  _active={{ background: 'transparent' }}
                  color={isOpen ? ' #344054' : '#98A2B3'}
                />
              </Box>
            ) : (
              <Text size="label-sm-default">{menuButtonText}</Text>
            )}
          </MenuButton>

          <MenuList
            boxShadow={
              '0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)'
            }
            marginTop={gap}
            data-testid="menu-list"
            py={'0px'}
            fontSize={'14px'}
            color="neutrals.900"
          >
            {menuHeader && (
              <HStack p={'16px 12px'}>
                <Avatar src={menuHeader?.profileUrl} width="40px" height="40px">
                  {indicator && <AvatarBadge boxSize="1em" bg="#00C408" />}
                </Avatar>

                <Box>
                  <Text size="label-xs-default" color="neutrals.900" mb="4px">
                    {menuHeader?.userName}
                  </Text>
                  <Text size="label-xs-default" color="neutrals.500">
                    {menuHeader?.emailAddress}
                  </Text>
                </Box>
              </HStack>
            )}
            <Divider orientation="horizontal" />
            <Menu>
              {() => {
                return <>{menuItems}</>;
              }}
            </Menu>
          </MenuList>
        </Box>
      )}
    </Menu>
  );
};

export default MenuDropdown;
