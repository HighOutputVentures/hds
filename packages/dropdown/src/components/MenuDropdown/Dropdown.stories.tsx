import {
  Avatar,
  AvatarBadge,
  Box,
  Flex,
  HStack,
  Text,
  VStack,
} from '@chakra-ui/react';
import {
  ExitIcon,
  SettingIcon,
  UserIcon,
  ZapIcon,
} from '@highoutput/hds-icons';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import ThemeProvider from '../../../../hds/src/components/ThemeProvider';
import Dropdown, { IMenuItemsGroup, IMenuItemsSingle } from './Dropdown';

const avatar = require('../../../../../assets/avatar.png');

export default {
  title: 'Components/Menu Dropdown',
  component: Dropdown,
} as ComponentMeta<typeof Dropdown>;

const itemsSingle: IMenuItemsSingle['items'] = [
  { label: 'View Profile', command: '⌘K->P', icon: <UserIcon /> },
  { label: 'Settings', command: '⌘S', icon: <SettingIcon /> },
  { label: 'Keyboard shortcuts', command: '?', icon: <ZapIcon /> },
];
const itemsGroup: IMenuItemsGroup['items'] = [
  [
    { label: 'View Profile', command: '⌘K->P', icon: <UserIcon /> },
    { label: 'Settings', command: '⌘S', icon: <SettingIcon /> },
    { label: 'Keyboard shortcuts', command: '?', icon: <ZapIcon /> },
  ],
  [{ label: 'Log out', icon: <ExitIcon />, command: '⌥⇧Q' }],
];

const Template: ComponentStory<typeof Dropdown> = (args) => {
  return (
    <ThemeProvider>
      <Flex
        justify={'space-around'}
        alignItems="start"
        flexDir="row"
        height={'500px'}
      >
        <VStack spacing="20px">
          <Text>Single</Text>
          <Dropdown
            items={itemsSingle}
            renderHeader={
              <HStack p={'16px 12px'}>
                <Avatar src={avatar} width="40px" height="40px">
                  <AvatarBadge boxSize="1em" bg="#00C408" />
                </Avatar>
                <Box>
                  <Text size="label-xs-default" color="neutrals.900" mb="4px">
                    Olivia Rhye
                  </Text>
                  <Text size="label-xs-default" color="neutrals.500">
                    olivia@untitledui.com
                  </Text>
                </Box>
              </HStack>
            }
            group={false}
            hasDivider
          >
            {({ onToggle }) => {
              return <Avatar src={avatar} onClick={onToggle} />;
            }}
          </Dropdown>
        </VStack>
        <VStack spacing="20px">
          <Text>Group</Text>
          <Dropdown
            items={itemsGroup}
            renderHeader={
              <HStack p={'16px 12px'}>
                <Avatar src={avatar} width="40px" height="40px">
                  <AvatarBadge boxSize="1em" bg="#00C408" />
                </Avatar>
                <Box>
                  <Text size="label-xs-default" color="neutrals.900" mb="4px">
                    Olivia Rhye
                  </Text>
                  <Text size="label-xs-default" color="neutrals.500">
                    olivia@untitledui.com
                  </Text>
                </Box>
              </HStack>
            }
            group
          >
            {({ onToggle }) => {
              return <Avatar src={avatar} onClick={onToggle} />;
            }}
          </Dropdown>
        </VStack>
      </Flex>
    </ThemeProvider>
  );
};

export const Default = Template.bind({});
