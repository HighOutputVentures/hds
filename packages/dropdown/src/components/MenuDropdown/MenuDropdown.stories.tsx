import {
  Divider,
  Flex,
  MenuGroup,
  MenuItem,
  Text,
  VStack,
} from '@chakra-ui/react';
import ThemeProvider from '@highoutput/hds/src/components/ThemeProvider';
import { ComponentMeta, ComponentStory } from '@storybook/react';
//@ts-ignore
import React from 'react';
import BoxIcon from '../Icon/BoxIcon';
import ExitIcon from '../Icon/ExitIcon';
import HelpIcon from '../Icon/HelpIcon';
import HomeIcon from '../Icon/HomeIcon';
import LayersTwoIcon from '../Icon/LayersTwoIcon';
import MessageSmileIcon from '../Icon/MessageSmileIcon';
import SettingIcon from '../Icon/SettingIcon';
import UserIcon from '../Icon/UserIcon';
import UserPlusIcon from '../Icon/UserPlusIcon';
import UsersIcon from '../Icon/UsersIcon';
import ZapIcon from '../Icon/ZapIcon';

import MenuDropdown from './MenuDropdown';
const avatar = require('../../../../../assets/avatar.png');
export default {
  title: 'Components/Menu Dropdown',
  component: MenuDropdown,
} as ComponentMeta<typeof MenuDropdown>;
const Template: ComponentStory<typeof MenuDropdown> = () => {
  const menuItems = [
    <MenuGroup>
      <MenuItem icon={<UserIcon />} command="⌘K->P">
        View Profile
      </MenuItem>
      <MenuItem icon={<SettingIcon />} command="⌘S">
        Settings
      </MenuItem>

      <MenuItem icon={<ZapIcon />} command="?">
        Keyboard shortcuts
      </MenuItem>
    </MenuGroup>,
    <Divider />,
    <MenuGroup>
      <MenuItem icon={<HomeIcon />} command="⌘K->C">
        Company profile
      </MenuItem>

      <MenuItem icon={<UsersIcon />} command="⌘K->T">
        Team
      </MenuItem>

      <MenuItem icon={<UserPlusIcon />} command="⌘K->T">
        Invite colleagues
      </MenuItem>
    </MenuGroup>,
    <Divider />,
    <MenuGroup>
      <MenuItem icon={<LayersTwoIcon />} command="⌘K->C">
        Changelog
      </MenuItem>
      <MenuItem icon={<MessageSmileIcon />} command="⌘K->S">
        Slack Community
      </MenuItem>
      <MenuItem icon={<HelpIcon />} command="⌘/">
        Support
      </MenuItem>
      <MenuItem icon={<BoxIcon />} command="⌘A">
        API
      </MenuItem>
    </MenuGroup>,
    <Divider />,
    <MenuItem
      icon={<ExitIcon />}
      command="⌥⇧Q"
      fontSize={'14px'}
      color="neutrals.900"
    >
      Log out
    </MenuItem>,
  ];
  return (
    <ThemeProvider>
      <Flex
        justify={'space-around'}
        alignItems="start"
        flexDir="row"
        height={'500px'}
      >
        <VStack spacing="20px">
          <Text fontSize={'18px'} fontWeight={500}>
            Profile dropdown type
          </Text>
          <MenuDropdown
            menuType={'profile'}
            profileUrl={avatar}
            menuHeader={{
              profileUrl: avatar,
              emailAddress: 'olivia@untitledui.com',
              userName: 'Olivia Rhye',
            }}
            menuItems={menuItems}
          />
        </VStack>
        <VStack spacing="20px">
          <Text fontSize={'18px'} fontWeight={500}>
            Kebab dropdown type
          </Text>
          <MenuDropdown
            menuType={'kebab'}
            menuHeader={{
              profileUrl: avatar,
              emailAddress: 'olivia@untitledui.com',
              userName: 'Olivia Rhye',
            }}
            menuItems={menuItems}
          />
        </VStack>
        <VStack spacing="20px">
          <Text fontSize={'18px'} fontWeight={500}>
            Button dropdown type
          </Text>
          <MenuDropdown
            menuType={'button'}
            menuHeader={{
              profileUrl: avatar,
              emailAddress: 'olivia@untitledui.com',
              userName: 'Olivia Rhye',
            }}
            menuButtonText="Account"
            menuItems={menuItems}
          />
        </VStack>
      </Flex>
    </ThemeProvider>
  );
};

export const Default = Template.bind({});

Default.args = {
  ...Default.args,
};
