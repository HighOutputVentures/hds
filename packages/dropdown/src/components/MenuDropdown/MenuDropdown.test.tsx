import { render, screen } from '@testing-library/react';
import React from 'react';
import MenuDropdown from './MenuDropdown';

import { RepeatIcon } from '@chakra-ui/icons';
import { MenuItem } from '@chakra-ui/react';
import ThemeProvider from '@highoutput/hds/src/components/ThemeProvider';

describe('Menu Dropdown Kebab Type Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <MenuDropdown
          menuHeader={{
            profileUrl: '',
            emailAddress: 'olivia@untitledui.com',
            userName: 'Olivia Rhye',
          }}
          menuItems={[
            <MenuItem
              icon={<RepeatIcon />}
              command="⌘⇧N"
              fontSize={'14px'}
              color="neutrals.900"
            >
              Open Closed Tab
            </MenuItem>,
          ]}
          menuType="kebab"
        />
      </ThemeProvider>
    );
  });

  it('Should render menu dropdown', async () => {
    const MenuKebab = await screen.findAllByTestId('menu-kebab');

    expect(MenuKebab).toHaveLength(1);
  });
});
describe('Menu Dropdown Button Type Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <MenuDropdown
          menuHeader={{
            profileUrl: '',
            emailAddress: 'olivia@untitledui.com',
            userName: 'Olivia Rhye',
          }}
          menuItems={[
            <MenuItem
              icon={<RepeatIcon />}
              command="⌘⇧N"
              fontSize={'14px'}
              color="neutrals.900"
            >
              Open Closed Tab
            </MenuItem>,
          ]}
          menuType="button"
          menuButtonText="Account"
        />
      </ThemeProvider>
    );
  });

  it('Should render menu dropdown', async () => {
    const MenuKebab = await screen.findAllByTestId('menu-button');

    expect(MenuKebab).toHaveLength(1);
  });
});
describe('Menu Dropdown Profile Type Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <MenuDropdown
          menuHeader={{
            profileUrl: '',
            emailAddress: 'olivia@untitledui.com',
            userName: 'Olivia Rhye',
          }}
          menuItems={[
            <MenuItem
              icon={<RepeatIcon />}
              command="⌘⇧N"
              fontSize={'14px'}
              color="neutrals.900"
            >
              Open Closed Tab
            </MenuItem>,
          ]}
          menuType="profile"
          profileUrl=""
        />
      </ThemeProvider>
    );
  });

  it('Should render menu dropdown', async () => {
    const MenuKebab = await screen.findAllByTestId('menu-profile');

    expect(MenuKebab).toHaveLength(1);
  });
});
