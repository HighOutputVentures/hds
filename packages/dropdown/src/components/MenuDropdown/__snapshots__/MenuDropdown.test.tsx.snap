// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded={false}
    aria-haspopup="menu"
    className="chakra-button chakra-menu__menu-button css-wzb3yu"
    data-testid="menu-profile"
    id="menu-button-:r1:"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <span
      className="css-xl71ch"
    >
      <span
        className="chakra-avatar css-sb6ugy"
      >
        <svg
          aria-label=" avatar"
          className="chakra-avatar__svg css-16ite8i"
          role="img"
          viewBox="0 0 128 128"
        >
          <path
            d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
            fill="currentColor"
          />
          <path
            d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
            fill="currentColor"
          />
        </svg>
      </span>
    </span>
  </button>,
  <div
    className="css-r6z5ec"
    style={
      Object {
        "inset": "0 auto auto 0",
        "minWidth": "max-content",
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    <div
      aria-orientation="vertical"
      className="chakra-menu__menu-list css-lfkiw7"
      data-testid="menu-list"
      id="menu-list-:r1:"
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.8) translateZ(0)",
          "transformOrigin": "var(--popper-transform-origin)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    >
      <div
        className="chakra-stack css-6aub0l"
      >
        <span
          className="chakra-avatar css-37of3x"
        >
          <svg
            aria-label=" avatar"
            className="chakra-avatar__svg css-16ite8i"
            role="img"
            viewBox="0 0 128 128"
          >
            <path
              d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
              fill="currentColor"
            />
            <path
              d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
              fill="currentColor"
            />
          </svg>
          <div
            className="chakra-avatar__badge css-13moskz"
          />
        </span>
        <div
          className="css-0"
        >
          <p
            className="chakra-text css-zju45d"
          >
            Olivia Rhye
          </p>
          <p
            className="chakra-text css-1kcm7zi"
          >
            olivia@untitledui.com
          </p>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-svjswr"
      />
      <button
        className="chakra-menu__menuitem css-1hsqdlt"
        id="menu-list-:r5:-menuitem-:r6:"
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <span
          className="chakra-menu__icon-wrapper css-15v20fh"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon chakra-menu__icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M10.319,4.936a7.239,7.239,0,0,1,7.1,2.252,1.25,1.25,0,1,0,1.872-1.657A9.737,9.737,0,0,0,9.743,2.5,10.269,10.269,0,0,0,2.378,9.61a.249.249,0,0,1-.271.178l-1.033-.13A.491.491,0,0,0,.6,9.877a.5.5,0,0,0-.019.526l2.476,4.342a.5.5,0,0,0,.373.248.43.43,0,0,0,.062,0,.5.5,0,0,0,.359-.152l3.477-3.593a.5.5,0,0,0-.3-.844L5.15,10.172a.25.25,0,0,1-.2-.333A7.7,7.7,0,0,1,10.319,4.936Z"
              />
              <path
                d="M23.406,14.1a.5.5,0,0,0,.015-.526l-2.5-4.329A.5.5,0,0,0,20.546,9a.489.489,0,0,0-.421.151l-3.456,3.614a.5.5,0,0,0,.3.842l1.848.221a.249.249,0,0,1,.183.117.253.253,0,0,1,.023.216,7.688,7.688,0,0,1-5.369,4.9,7.243,7.243,0,0,1-7.1-2.253,1.25,1.25,0,1,0-1.872,1.656,9.74,9.74,0,0,0,9.549,3.03,10.261,10.261,0,0,0,7.369-7.12.251.251,0,0,1,.27-.179l1.058.127a.422.422,0,0,0,.06,0A.5.5,0,0,0,23.406,14.1Z"
              />
            </g>
          </svg>
        </span>
        <span
          style={
            Object {
              "flex": 1,
              "pointerEvents": "none",
            }
          }
        >
          Open Closed Tab
        </span>
        <span
          className="chakra-menu__command css-dt0657"
        >
          ⌘⇧N
        </span>
      </button>
    </div>
  </div>,
  <span />,
]
`;
